nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/download/DownloadServiceImpl.java

```diff
@@ -336,6 +339,15 @@ protected void handleDownload(HttpServletRequest req, HttpServletResponse resp,
             try (CoreSession session = CoreInstance.openCoreSession(downloadBlobInfo.repository)) {
                 DocumentRef docRef = new IdRef(downloadBlobInfo.docId);
                 if (!session.exists(docRef)) {
+                    // Authenticate first, if the user is anonymous
```

Comment:
Please clarify by saying something like "If the user is anonymous, send a security exception to force authentication with a real user".